src/autoloads/EnemyUtil.gd

```diff
@@ -1,3 +1,13 @@
 extends Node
 
-var aggro_array : Array[CharacterBody2D]
+## Array that stores every [Node2D] that is currently aggrod on the [Player]
+var aggro_array : Array[Node2D]
+
+## Function that should be used whenever a [Node2D] aggros onto the [Player] character, so that
+## we can keep track of it.
+func begin_aggro(aggresor: Node2D):
+	aggro_array.append(aggresor)
```

Comment:
this unconditional append could lead to a bug with how remove is appended - here is where using a dictionary would shine, as recurring adds wouldn't have the possibility of leaking. i'd appreciate *at least* an assert to stop from the same aggressor registering twice, though i don't think looping through the array once for begin aggro is that performance heavy